packages/js/src/ui/context/InboxContext.tsx

```diff
@@ -0,0 +1,74 @@
+import { Accessor, createContext, createSignal, ParentProps, useContext } from 'solid-js';
+import { NotificationFilter } from '../../types';
+import { NotificationStatus, Tab } from '../types';
+
+type InboxContextType = {
+  setStatus: (status: NotificationStatus) => void;
+  status: Accessor<NotificationStatus>;
+  filter: Accessor<NotificationFilter>;
+  tabs: Accessor<Array<Tab>>;
+  activeTab: Accessor<string>;
+  setActiveTab: (tab: string) => void;
+};
+
+const InboxContext = createContext<InboxContextType | undefined>(undefined);
+
+const STATUS_TO_FILTER: Record<NotificationStatus, NotificationFilter> = {
+  [NotificationStatus.UNREAD_READ]: { archived: false },
+  [NotificationStatus.UNREAD]: { read: false },
+  [NotificationStatus.ARCHIVED]: { archived: true },
+};
+
+type InboxProviderProps = ParentProps<{
+  tabs: Array<Tab>;
+}>;
+
+export const InboxProvider = (props: InboxProviderProps) => {
```

Comment:
The `InboxProvider` takes the `tabs` as the props passed, calculates the active tab, and builds the initial filter.